frappe/integrations/utils.py

```diff
@@ -21,15 +21,15 @@ def make_request(method: str, url: str, auth=None, headers=None, data=None, json
 		)
 		response.raise_for_status()
 
-		content_type = response.headers.get("content-type")
-		if content_type == "text/plain; charset=utf-8":
-			return parse_qs(response.text)
-		elif content_type.startswith("application/") and content_type.split(";")[0].endswith("json"):
-			return response.json()
-		elif response.text:
-			return response.text
-		else:
-			return
+		# Check whether the response has a content-type, before trying to check what it is
+		if content_type := response.headers.get("content-type"):
+			if content_type == "text/plain; charset=utf-8":
+				return parse_qs(response.text)
+			elif content_type.startswith("application/") and content_type.split(";")[0].endswith("json"):
+				return response.json()
+			elif response.text:
+				return response.text
```

Comment:
@akhilnarang should this be outside the `content_type` block? Since the old code would return `response.text` as a fallback

Reply:
I'm not sure that's true, since I think the `requests` client will try to [guess](https://requests.readthedocs.io/en/latest/api/#requests.Response.text) the response type and encoding. But if it's low impact, then I guess this is fine too.

Reply:
That just mentions encoding, not content-type right?

Will try to see if it breaks any valid use cases and change accordingly anyway.